geomesa-fs/geomesa-fs-tools/src/main/scala/org/locationtech/geomesa/fs/tools/compact/ParquetCompactionJob.scala

```diff
@@ -0,0 +1,298 @@
+    FileOutputFormat.setOutputPath(job, tempPath.getOrElse(dsPath))
+
+    // Parquet Options
+    val summaryLevel = Option(sft.getUserData.get(ParquetOutputFormat.JOB_SUMMARY_LEVEL).asInstanceOf[String])
+      .getOrElse(ParquetOutputFormat.JobSummaryLevel.NONE.toString)
+    job.getConfiguration.set(ParquetOutputFormat.JOB_SUMMARY_LEVEL, summaryLevel)
+    Command.user.info(s"Parquet metadata summary level is $summaryLevel")
+
+    val compression = Option(sft.getUserData.get(ParquetOutputFormat.COMPRESSION).asInstanceOf[String])
+      .map(CompressionCodecName.valueOf)
+      .getOrElse(CompressionCodecName.SNAPPY)
+    ParquetOutputFormat.setCompression(job, compression)
+    Command.user.info(s"Parquet compression is $compression")
+
+    // More Parquet config
+    ParquetOutputFormat.setWriteSupportClass(job, classOf[SimpleFeatureWriteSupport])
+    ParquetJobUtils.setSimpleFeatureType(job.getConfiguration, sft)
+
+    // Save the existing files so we can delete them afterwards
+    // Be sure to filter this based on the input partitions
+    val existingDataFiles = ds.storage.listPartitions(typeName)
+      .intersect(partitions).flatMap(ds.storage.getPaths(typeName, _)).toList
+
+    Command.user.info("Submitting job - please wait...")
+    job.submit()
+    Command.user.info(s"Tracking available at ${job.getStatus.getTrackingUrl}")
+
+    def mapCounters = Seq(("mapped", written(job)), ("failed", failed(job)))
+
+    val stageCount = if (tempPath.isDefined) { "2" } else { "1" }
+
+    while (!job.isComplete) {
+      if (job.getStatus.getState != JobStatus.State.PREP) {
+        val mapProgress = job.mapProgress()
+        if (mapProgress < 1f) {
+          statusCallback(s"Map (stage 1/$stageCount): ", mapProgress, mapCounters, done = false)
+        } else {
+          statusCallback(s"Map (stage 1/$stageCount): ", mapProgress, mapCounters, done = true)
+          statusCallback.reset()
+        }
+      }
+      Thread.sleep(1000)
+    }
+
+    val res = (written(job), failed(job))
+
+    val ret = job.isSuccessful &&
+      tempPath.forall(tp => ParquetJobUtils.distCopy(tp, dsPath, sft, job.getConfiguration, statusCallback)) && {
+
+        Command.user.info("Removing old files")
+        val fs = ds.root.getFileSystem(job.getConfiguration)
+        existingDataFiles.foreach(o => fs.delete(new Path(o), false))
+        Command.user.info(s"Removed ${existingDataFiles.size} files")
+
+        Command.user.info("Updating metadata")
+        // We sleep here to allow a chance for S3 to become "consistent" with its storage listings
+        Thread.sleep(5000)
+        ds.storage.updateMetadata(typeName)
+        Command.user.info("Metadata Updated")
+        true
+      }
+
+    if (!ret) {
+      Command.user.error(s"Job failed with state ${job.getStatus.getState} due to: ${job.getStatus.getFailureInfo}")
+    }
+
+    res
+  }
+
+  override def inputFormatClass: Class[_ <: FileInputFormat[_, SimpleFeature]] = null
```

Comment:
[minor] in the future we should probably extract out `readLibJars` and `defaultSearchPath` into a new trait and extend that instead of `AbstractIngestJob`

Reply:
definitely. i almost went down that road but then decided just to throw it up since this isn't really an "ingest" job